src/app/lab/lab.component.ts

```diff
@@ -68,94 +85,71 @@ export class LabComponent implements OnInit, OnDestroy {
   public onCitySelected(value: any) {
     return (value) => {
       this.project.city = value;
-      this.chartService.updateCity(value);
+      this.projectService.update(this.project);
     };
   }
 
+  public chartSettingChanged() {
+    this.projectService.update(this.project);
+  }
+
   // unselect all model checkboxes when option to use all models selected
   public clearClimateModels() {
-    _.each(this.project.models, function(model) {
-      model.selected = false;
-    });
+    this.climateModels.forEach(model => model.selected = false);
   }
 
-  // deselect option to use all models when one or more model is checked; will also enable the
-  // checkbox to use all models (use-all checkbox can only be deselected by selecting a model)
-  public deselectUseAllModels() {
-    this.project.allModels = false;
+  public isModalUpdateButtonDisabled() {
+      return this.climateModels.filter(model => model.selected).length === 0;
   }
 
-  public updateClimateModels() {
-    let models = this.getSelectedClimateModels();
-    if (!models.length) {
-      this.project.allModels = true;
-    }
-    this.chartService.updateClimateModels(models);
+  public selectAllClimateModels() {
+    this.climateModels.forEach(model => model.selected = true);
+  }
+
+  public updateProjectClimateModels() {
+    this.project.models = this.climateModels.filter(model => model.selected);
+    this.projectService.update(this.project);
   }
 
   public updateScenario(scenario: Scenario) {
     this.project.scenario = scenario;
-    this.chartService.updateScenario(scenario.name);
+    this.projectService.update(this.project);
   }
 
   public removeChart(chart: Chart) {
-    this.chartService.removeChart(chart);
-    this.project.charts = this.chartService.get();
+    this.project.charts = this.project.charts.filter(c => c !== chart);
+    this.projectService.update(this.project);
   }
 
-  // subscribe to list of available models from API endpoint
-  getAvailableClimateModels() {
-    this.climateModelService.list().subscribe(data => {
-      this.project.models = data;
-    });
+  public indicatorSelected(indicator: Indicator) {
+    let chart = new Chart({indicator: indicator});
+    this.project.charts.push(chart);
+    this.projectService.update(this.project);
   }
 
-  // Returns names of the models currently selected by the user
-  getSelectedClimateModels(): string[] {
-    return this.project.models.filter(function(model) {
-      return model.selected;
-    }).map(function(model) {
-      return model.name;
+  // subscribe to list of available models from API endpoint
+  private getClimateModels() {
+    this.climateModelService.list().subscribe(data => {
+      this.climateModels = data;
+
+      // Initialize 'selected' attributes with models in project
+      if (this.project.models.length === 0) {
+        this.selectAllClimateModels();
+        this.updateProjectClimateModels();
+      } else {
+        this.project.models.forEach(projectModel => {
```

Comment:
Just brute forced this update/comparison. Always a very small dataset so I don't think its worth it to optimize.